### PR TITLE
Set up ESLint for JSDoc comments

### PR DESCRIPTION
**Pull request by koddsson**
I really want to explore both better typing and possible automatic generation of API docs since the JSDoc comments that exist in the codebase are of such high quality.

This PR introduces ESLint and `eslint-plugin-jsdoc` to start linting JSDoc comment in effort to make sure that all of the comments stay high quality as we refactor the code

This PR can be followed up by more changes to the rules we apply as well as some explorations into API documentation and stricter type-checking.

### CODE REVIEW

File: eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import jsdoc from "eslint-plugin-jsdoc";
+
+export default [
+  jsdoc.configs["flat/recommended"],
```
Comment (43081j) [inline eslint.config.js:4]: while you're in there, can you maybe also enable eslint's recommended rule set? im surprised we didn't already have eslint

if it ends up with too big of a change, you could do a PR to add eslint + recommended, then a separate one for jsdoc on top

File: lib/chai/core/assertions.js [hunk 2]
```diff
@@ -2116,6 +2110,13 @@ function assertOwnPropertyDescriptor (name, descriptor, msg) {
 Assertion.addMethod('ownPropertyDescriptor', assertOwnPropertyDescriptor);
 Assertion.addMethod('haveOwnPropertyDescriptor', assertOwnPropertyDescriptor);
 
+/**
+ *
```
Comment (43081j) [inline lib/chai/core/assertions.js:2114]: maybe add a description? "Flags that the `length` property should be asserted in this chain"

**Review by 43081j — APPROVED** [review]
looks good, though it may be worth marking optional params as such or just wait for my typescript conversion to land

e.g. `@param {string=} iAmAnOptionalParam`

**Comment by koddsson** [comment]
lol, maybe I shouldn't have had it on auto-merge :D 

I'll make a follow up PR to apply some of your suggestions @43081j

**Pull request merged**